include/sparrow/utils/format.hpp

```diff
@@ -0,0 +1,204 @@
+    {
+        size_t max_width = 0;
+        for (const auto& value : data)
+        {
+            max_width = std::max(max_width, std::format("{}", value).size());
+        }
+        return max_width;
+    }
+
+    template <RangeOfRanges Columns>
+    constexpr std::vector<size_t> columns_widths(const Columns& columns)
+    {
+        std::vector<size_t> widths;
+        widths.reserve(std::ranges::size(columns));
+        for (const auto& col : columns)
+        {
+            widths.push_back(max_width(col));
+        }
+        return widths;
+    }
+
+    template <typename OutputIt, std::ranges::input_range Widths, std::ranges::input_range Values>
+        requires(std::same_as<std::ranges::range_value_t<Widths>, size_t>)
+    constexpr void
+    to_row(OutputIt out, const Widths& widths, const Values& values, std::string_view separator = "|")
+    {
+        SPARROW_ASSERT_TRUE(std::ranges::size(widths) == std::ranges::size(values))
+        if (std::ranges::size(values) == 0)
+        {
+            return;
+        }
+        auto value_it = values.begin();
+        auto width_it = widths.begin();
+        for (size_t i = 0; i < std::ranges::size(values); ++i)
+        {
+            const std::string fmt = std::format("{}{{:>{}}}", separator, *width_it);
+            const auto& value = *value_it;
+            std::vformat_to(out, fmt, std::make_format_args(value));
+            ++value_it;
+            ++width_it;
+        }
+        std::format_to(out, "{}", separator);
+    }
+
+    template <typename OutputIt>
+    constexpr void
+    horizontal_separator(OutputIt out, const std::vector<size_t>& widths, std::string_view separator = "-")
+    {
+        if (std::ranges::size(widths) == 0)
+        {
+            return;
+        }
+        const size_t count = std::ranges::size(widths) + 1 + std::reduce(widths.begin(), widths.end());
+        std::format_to(out, "{}", std::string(count, separator[0]));
+    }
+
+    template <std::ranges::input_range Headers, RangeOfRanges Columns, typename OutputIt>
+        requires(std::convertible_to<std::ranges::range_value_t<Headers>, std::string>)
+    constexpr void to_table_with_columns(OutputIt out, const Headers& headers, const Columns& columns)
+    {
+        const size_t column_count = std::ranges::size(columns);
+        SPARROW_ASSERT_TRUE(std::ranges::size(headers) == column_count);
+        if (column_count == 0)
+        {
+            return;
+        }
+
+        // columns lenght must be the same
+        if (column_count > 0)
+        {
+            for (auto it = columns.begin() + 1; it != columns.end(); ++it)
+            {
+                SPARROW_ASSERT_TRUE(std::ranges::size(columns[0]) == std::ranges::size(*it));
+            }
+        }
+
+        std::vector<size_t> widths = columns_widths(columns);
+        
+        // max with names
+        for (size_t i = 0; i < std::ranges::size(headers); ++i)
+        {
+            #if defined(__clang__)
```

Comment:
Would a `static_cast` avoid the warning here? I think it would be easier to read. If not, then maybe a single set of pragma enclosing the whole function would probably improve the readability.

Reply:
No because, as headers is a template argument, depend ending of its type, `std::ranges::size(headers)` will not return the same type, in one case we have teh warning about the sign conversion, in the other a warning about useless static_cast.

Reply:
I still think a single set of pragma enclosing the whole function would make it easier to read.

Reply:
done